src/AssemblerReads.cpp

```diff
@@ -183,14 +184,90 @@ void Assembler::writeReadsSummary()
         csv << double(vertexCount) / double(markerCount) << ",";
 
         // Oxford Nanopore standard metadata.
-        csv << reads.getMetaData(readId, "runid") << ",";
-        csv << reads.getMetaData(readId, "sampleid") << ",";
-        csv << reads.getMetaData(readId, "read") << ",";
-        csv << reads.getMetaData(readId, "ch") << ",";
-        csv << reads.getMetaData(readId, "start_time") << ",";
+        csv << reads->getMetaData(readId, "runid") << ",";
+        csv << reads->getMetaData(readId, "sampleid") << ",";
+        csv << reads->getMetaData(readId, "read") << ",";
+        csv << reads->getMetaData(readId, "ch") << ",";
+        csv << reads->getMetaData(readId, "start_time") << ",";
 
         // End of the line for this read.
         csv << "\n";
-     }
+    }
+}
+
+
+void Assembler::adjustCoverage(uint64_t minReadLength, uint64_t desiredCoverage) {
+    if (desiredCoverage == 0) {
+        // Not specified. Skip.
+        return;
+    }
+
+    // Need to compute histograms to adjust coverage.
+    reads->computeReadLengthHistogram();
+
+    cout << timestamp << "Adjusting for desired coverage." << endl;
+    cout << "Desired Coverage: " << desiredCoverage << endl;
+    uint64_t cumulativeBaseCount = reads->getTotalBaseCount();
+
+    if (desiredCoverage > cumulativeBaseCount) {
+        throw runtime_error(
+            "With a Reads.minReadLength of " + to_string(minReadLength) + ","
+            "the total available coverage (" + to_string(cumulativeBaseCount) +
+            ") is lesser than the desired coverage (" + to_string(desiredCoverage) +
+            "). Try reducing Reads.minReadLength if appropriate or get more coverage."
+        );
+    }
+
+    uint64_t newMinReadLength = std::numeric_limits<uint64_t>::max();
+
+    const auto& binnedHistogram = reads->getBinnedHistogram();
```

Comment:
Why do it on the binned histogram instead of the complete histogram, which is also available? As written, it will always use a multiple of 1 Kb as the new read length cutoff.